### PR TITLE
(CONT-1028) puppetlabs/stdlib: Allow 9.x

### PR DESCRIPTION
**Pull request by bastelfreak**
this is a resubmission of https://github.com/puppetlabs/puppetlabs-apt/pull/1111

### CODE REVIEW

**Review by LukasAud — APPROVED** [review]
LGTM. Thanks for this contribution!

**Comment by bastelfreak** [comment]
@LukasAud thanks for the quick merge! Is it possible to get a new release soon?

File: metadata.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "dependencies": [
     {
       "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 4.16.0 < 9.0.0"
+      "version_requirement": ">= 4.16.0 < 10.0.0"
```
Comment (smortex) [inline metadata.json:13]: Hum, `stdlib::ensure_packages` appeared in `9.0.0` (where `merge` is also available as `stdlib::merge`).  Not sure if we should bump the minimum version to 9.0.0 or rather use the deprecated functions.

Comment (bastelfreak) [reply metadata.json:13]: I prefer to do it now and rip out all the legacy function calls. Otherwise we need to touch the codebase in the future as well.

Comment (smortex) [reply metadata.json:13]: ACK! I opened #1114 to update the minimum version requirement.

**Comment by LukasAud** [comment]
@bastelfreak We are working on releases. However, due to some recent maintenance, there is a lengthy task that we need to address before we can cut a release for each module. We will try to release as soon as possible.

**Pull request merged**